tests/python/pants_test/engine/test_rules.py

```diff
@@ -201,6 +202,32 @@ def goal_rule_returning_a_non_goal() -> int:
         return 0
 
 
+def fmt_task_func(func: Callable) -> str:
+  """Generate the str for a Rust Func, which is how Rust refers to `@rule`s."""
+  return f"{func.__module__}:{func.__code__.co_firstlineno}:{func.__name__}"
+
+
+def fmt_rule(
+  rule: Callable, *, gets: Optional[List[Tuple[str, str]]] = None,
+) -> str:
+  """Generate the str that the engine will use for the rule.
+
+  This is very tightly coupled with the engine - any changes to Rust's display of rules will
```

Comment:
I like this reasoning, but is this lengthy comment necessary (sorry! keep it if you want!)? Is the idea to say "this method is tightly coupled to Display impls in the rust codebase", or something else (not sure of the second sentence)?

Reply:
A justification for why it exists, I suppose. But I think that's unnecessary given how complex this function is - I don't imagine anyone trying to remove this and hardcode it haha.